bpf/bpf_lxc.c

```diff
@@ -1089,7 +1093,7 @@ __section_tail(CILIUM_MAP_POLICY, LXC_ID) int handle_policy(struct __sk_buff *sk
 	return ret;
 }
 
-#ifdef ENABLE_NAT46
+#if defined(ENABLE_NAT46) && !defined(DATAPATH_IPVLAN)
```

Comment:
This can stay as-is if we undefine `ENABLE_NAT46`

Reply:
Agree, I'll improve this logic. NAT64 code doesn't work with ipvlan since if we rewrite addresses from v4 to v6 and vice versa, then ipvlan internal fdb logic won't find local target container anymore. If we ever need this feature here we could i) push the skbs to the phys device, and do reverse translation from there to then push it back in so that lookup would work, or we could add a ipvlan bpf operation mode to have custom lookups (or even policy moved into it).

Reply:
I've changed it to error out if a user tries to flip the ENABLE_NAT46 via `cilium endpoint config`.